### PR TITLE
Integrated 771

### PR DESCRIPTION
**Pull request by Abdaaf**
Database v3.5

### CODE REVIEW

File: Tests/Controller/DefaultControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (marijnotte) [inline Tests/Controller/DefaultControllerTest.php]: This seems to be an example file, which should not be commited

File: Controller/DefaultController.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (marijnotte) [inline Controller/DefaultController.php]: What does this controller do?

File: Resources/config/routing.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<routes xmlns="http://symfony.com/schema/routing"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/routing http://symfony.com/schema/routing/routing-1.0.xsd">
+
+    <route id="integrated_subscription_homepage" path="/hello/{name}">
```
Comment (marijnotte) [inline Resources/config/routing.xml]: Seems to be an example route?

File: Resources/views/Default/index.html.twig [hunk 4]
```diff
@@ -0,0 +1 @@
+Hello {{ name }}!
```
Comment (marijnotte) [inline Resources/views/Default/index.html.twig]: What does this do?

File: Controller/DefaultController.php (hunk 2 above)
Comment (johanliefers) [inline Controller/DefaultController.php]: Missing integrated copyright block and author tags, this counts for every document

File: Entity/Subscription.php [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+ */
+
+namespace Integrated\Bundle\SubscriptionBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * Integrated\Bundle\SubscriptionBundle\Entity\Subscription
+ *
+ * @ORM\Entity()
```
Comment (johanliefers) [inline Entity/Subscription.php]: Mapping should be in a seperate xml document. Check userbundle for example

File: Entity/Subscription.php [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+ * @ORM\Table(name="subscription", indexes={@ORM\Index(name="fk_Subscription_SubscriptionType1_idx", columns={"`type`"})})
```
Comment (johanliefers) [inline Entity/Subscription.php]: why make such a long index name?

File: Entity/Subscription.php [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+     * @ORM\JoinColumn(name="`type`", referencedColumnName="id", nullable=false)
+     */
+    protected $subscriptionType;
+
+    public function __construct()
+    {
+    }
+
+    /**
+     * Set the value of id.
```
Comment (johanliefers) [inline Entity/Subscription.php]: unnecessary addition, you only need to add a description if there is some logic that needs explaination. In this case it is clear this function sets the value of id. This counts for all the getters and setters without any custom logic

File: Entity/Subscription.php [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+     */
+    public function getStatus()
+    {
+        return $this->status;
+    }
+
+    /**
+     * Set SubscriptionType entity (many to one).
+     *
+     * @param \Integrated\Bundle\SubscriptionBundle\Entity\SubscriptionType $subscriptionType
```
Comment (johanliefers) [inline Entity/Subscription.php]: Unnecessary to use the complete namespace

File: Entity/Subscription.php [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+     * Get SubscriptionType entity (many to one).
+     *
+     * @return \Integrated\Bundle\SubscriptionBundle\Entity\SubscriptionType
+     */
+    public function getSubscriptionType()
+    {
+        return $this->subscriptionType;
+    }
+
+    public function __sleep()
```
Comment (johanliefers) [inline Entity/Subscription.php]: why do you add this magic function?

File: Entity/Subscription.php [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+        return $this->status;
+    }
+
+    /**
+     * Set SubscriptionType entity (many to one).
+     *
+     * @param \Integrated\Bundle\SubscriptionBundle\Entity\SubscriptionType $subscriptionType
+     * @return \Integrated\Bundle\SubscriptionBundle\Entity\Subscription
+     */
+    public function setSubscriptionType(SubscriptionType $subscriptionType = null)
```
Comment (johanliefers) [inline Entity/Subscription.php]: subscriptiontype is not nullable, so why make it default null?

File: Entity/SubscriptionTypeVariant.php [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+
+    /**
+     * @ORM\Column(type="string", nullable=true)
+     */
+    protected $periodtype;
+
+    /**
+     * @ORM\Column(type="integer", nullable=true)
+     */
+    protected $periodlentgh;
```
Comment (johanliefers) [inline Entity/SubscriptionTypeVariant.php]: use camelCase, also check other defined variables.

File: Entity/SubscriptionTypeVariant.php [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+
+    /**
+     * @ORM\ManyToOne(targetEntity="SubscriptionType", inversedBy="subscriptionTypeVariants")
+     * @ORM\JoinColumn(name="subscription_type_id", referencedColumnName="id")
+     */
+    protected $subscriptionType;
+
+    public function __construct()
```
Comment (johanliefers) [inline Entity/SubscriptionTypeVariant.php]: no need for constructor if it is empty

File: Entity/VatCountry.php [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+
+    /**
+     * Add VatType entity to collection (one to many).
+     *
+     * @param \Integrated\Bundle\SubscriptionBundle\Entity\VatType $vatType
+     * @return \Integrated\Bundle\SubscriptionBundle\Entity\VatCountry
+     */
+    public function addVatType(VatType $vatType)
+    {
+        $this->vatTypes[] = $vatType;
```
Comment (johanliefers) [inline Entity/VatCountry.php]: use the add function of the collection instead

File: LICENSE [hunk 14]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2015-2016 e-Active B.V.
+Copyright (c) 2013-2014 e-Active B.V.
```
Comment (johanliefers) [inline LICENSE:1]: check date

File: README.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
+# IntegratedBlockBundles #
+This bundle provides block management
```
Comment (johanliefers) [inline README.md]: Readme is of block bundle, you should create one for the subscription bundle

File: Resources/config/services.xml [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <!--
```
Comment (johanliefers) [inline Resources/config/services.xml]: Plz remove test code instead of comment it out

File: composer.json [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+    ],
+    "minimum-stability": "dev",
+    "prefer-stable": true,
+    "require": {
+        "php": ">=5.4",
+        "symfony/symfony": ">=2.4",
+        "symfony/framework-bundle": ">=2.2",
+        "sensio/framework-extra-bundle": "~2.3",
+        "doctrine/mongodb": ">=1.0.0",
+        "doctrine/mongodb-odm": "*",
```
Comment (johanliefers) [inline composer.json]: why do you require odm and mongo? I think most requirements stated here are actually not really required, plz check

File: Entity/Subscription.php [hunk 18]
```diff
@@ -0,0 +1,211 @@
+<?php
+
```
Comment (johanliefers) [inline Entity/Subscription.php]: Plz remove the auto generated comment and include the standard integrated copyright above every file

File: Entity/Subscription.php [hunk 19]
```diff
@@ -0,0 +1,211 @@
...
+namespace Integrated\Bundle\SubscriptionBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * Integrated\Bundle\SubscriptionBundle\Entity\Subscription
+ */
+class Subscription
+{
+    protected $id;
```
Comment (johanliefers) [inline Entity/Subscription.php]: Not all annotations should be deleted, plz keep the @var tag

File: Entity/SubscriptionWall.php [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+{
+    protected $id;
+
+    protected $name;
+
+    protected $teaser;
+
+    protected $disabled;
+
+    protected $freetier;
```
Comment (johanliefers) [inline Entity/SubscriptionWall.php]: What does this variable contain (I don't know the word)?

File: README.md [hunk 21]
```diff
@@ -0,0 +1,49 @@
+# IntegratedSubscriptionWallBundles #
```
Comment (johanliefers) [inline README.md]: should be IntegratedSubscriptionWallBundle

File: README.md [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+            // ...
+            new Integrated\Bundle\SubscriptionBundle\IntegratedSubscriptionBundle(),
+            // ...
+        );
+    }
+
+### Import the routing ###
+
+    # app/config/routing.yml
+    integrated_block:
```
Comment (johanliefers) [inline README.md]: should be integrated_subscription

File: Resources/config/doctrine/Subscription.orm.xml [hunk 23]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
+  <entity name="Integrated\Bundle\SubscriptionBundle\Entity\Subscription" table="subscription">
```
Comment (johanliefers) [inline Resources/config/doctrine/Subscription.orm.xml]: more white lines makes it more readable, I would suggest a new white line after every defined ORM variable

File: DependencyInjection/IntegratedSubscriptionExtension.php [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function load(array $configs, ContainerBuilder $container)
+    {
+        $configuration = new Configuration();
+        $config = $this->processConfiguration($configuration, $configs);
+
+        $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $loader->load('services.xml');
```
Comment (johanliefers) [inline DependencyInjection/IntegratedSubscriptionExtension.php:33]: not really necessary if you have no services yet, but you can leave it here for now, assuming that you will make some services in the future

File: Entity/WallChannel.php [hunk 25]
```diff
@@ -0,0 +1,89 @@
...
+ */
+class WallChannel
+{
+    protected $wall;
+
+    protected $channel;
+
+    protected $subscriptionWall;
+
+    public function __construct()
```
Comment (johanliefers) [inline Entity/WallChannel.php]: remove empty constructor

File: composer.json [hunk 26]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    ],
+    "minimum-stability": "dev",
+    "prefer-stable": true,
+    "require": {
+        "php": ">=5.4",
+        "symfony/symfony": ">=2.4",
+        "symfony/framework-bundle": ">=2.2",
+        "sensio/framework-extra-bundle": "~2.3",
+        "doctrine/mongodb": ">=1.0.0",
```
Comment (johanliefers) [inline composer.json]: why do you require odm and mongo? I think most requirements stated here are actually not really required, plz check

**Pull request merged**